src/OmnichannelChatSDK.ts

```diff
@@ -1208,19 +1654,34 @@ class OmnichannelChatSDK {
     }
 
     private resolveChatAdapterUrl(protocol: string): string {
-        if (protocol !== ChatAdapterProtocols.IC3) {
+        const supportedChatAdapterProtocols = [ChatAdapterProtocols.ACS, ChatAdapterProtocols.IC3];
+        if (protocol && !supportedChatAdapterProtocols.includes(protocol as string)) {
             throw new Error(`ChatAdapter for protocol ${protocol} currently not supported`);
         }
 
-        if (this.chatSDKConfig.chatAdapterConfig && 'webChatIC3AdapterCDNUrl' in this.chatSDKConfig.chatAdapterConfig) {
-            return this.chatSDKConfig.chatAdapterConfig.webChatIC3AdapterCDNUrl as string;
-        }
+        if (protocol === ChatAdapterProtocols.ACS || this.liveChatVersion === LiveChatVersion.V2) {
```

Comment:
shouldn't this be &&? Will there be any case where protocol is ACS and liveChatVersion in V1?

Reply:
I added `||` for more flexibility. Using `createChatAdapter()` without any `protocol` parameter (will be set as `null` by default), will return the right adapter based on `this.liveChatVersion`. You can pass a parameter such as `ACS` like `createChatAdapter('ACS')` to return a adapter based on the protocol specified.